Profanity with python plugins won't build on Gentoo (conflict of _XOPEN_SOURCE)

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
Profanity builds.

## Current Behavior
```
gcc -DHAVE_CONFIG_H -I. -I./src  -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -I/usr/include/ncursesw  -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi-3.2.1/include -pthread -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid    -I/usr/include/python3.6m  -DTHEMES_PATH="\"/usr/local/share/profanity/themes\"" -DICONS_PATH="\"/usr/local/share/profanity/icons\""  -Wall -Wno-deprecated-declarations -Wunused -Werror -I./src -g -O2 -pthread -I/usr/include/libxml2  -MT src/plugins/python_plugins.o -MD -MP -MF $depbase.Tpo -c -o src/plugins/python_plugins.o src/plugins/python_plugins.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from /usr/include/python3.6m/Python.h:8,
                 from src/plugins/python_plugins.c:35:
/usr/include/python3.6m/pyconfig.h:1493: error: "_XOPEN_SOURCE" redefined [-Werror]
 #define _XOPEN_SOURCE 700

<command-line>: note: this is the location of the previous definition
cc1: all warnings being treated as errors
```

## Possible Solution
```C
/* just before #include <Python.h> */
#undef _XOPEN_SOURCE
```

## Steps to Reproduce (for bugs)
1. ./bootstrap.sh && configure && make
2. Python plugins are enabled

## Context
The problem is that pyconfig.h (included in Python.h) defines `_XOPEN_SOURCE` unconditionally. At the same time, pkg-config defines the same macro for ncursesw:
```
# cat /usr/lib64/pkgconfig/ncursesw.pc | grep Cflags
Cflags:  -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -I${includedir}
```
As result, `_DEFAULT_SOURCE` defined in command line and then redefined in pyconfig.h.

This is not Profanity issue itself, but I think we can make Profanity build out of box somehow.

## Environment
* Master (commit b3ed21366879c53792277c069766e2d7f6cd5c5b)
* Gentoo Linux
* sys-libs/ncurses-6.1_p20181020
* dev-lang/python-3.6.5

--- a/src/plugins/python_api.c
+++ b/src/plugins/python_api.c
@@ -32,6 +32,7 @@
  *
  */
 
+#undef _XOPEN_SOURCE
 #include <Python.h>
 
 #include "config.h"

--- a/src/plugins/python_plugins.c
+++ b/src/plugins/python_plugins.c
@@ -32,6 +32,7 @@
  *
  */
 
+#undef _XOPEN_SOURCE
 #include <Python.h>
 
 #include "log.h"

